File: ruby/resistor-color-duo/resistor_color_duo.rb
module ResistorColorDuo
  COLORS = %w[black brown red orange yellow green blue violet grey white]

  def self.single_value(color)
    COLORS.find_index(color)
  end
  
  def self.value(colors)
    single_value(colors[0]) * 10 + single_value(colors[1])
  end
end

File: java/resistor-color-duo/_iterations/1/src/main/java/ResistorColorDuo.java
import java.util.Arrays;
import java.util.List;

class ResistorColorDuo {
    private static final List<String> COLORS = Arrays.asList("black", "brown", "red", "orange", "yellow",
            "green", "blue", "violet", "grey", "white");

    int value(String[] colors) {
        return COLORS.indexOf(colors[0]) * 10 + COLORS.indexOf(colors[1]);
    }
}
